packages/ag-charts-community/src/chart/interaction/cursorManager.ts

```diff
@@ -1,39 +1,17 @@
-interface CursorState {
-    style: string;
-}
+import { StateTracker } from '../../util/state-tracker';
 
 /**
  * Manages the cursor styling for an element. Tracks the requested styling from distinct
  * dependents and handles conflicting styling requests.
  */
 export class CursorManager {
-    private readonly states: Record<string, CursorState> = {};
-    private readonly element: HTMLElement;
+    private readonly stateTracker = new StateTracker('default');
 
-    public constructor(element: HTMLElement) {
-        this.element = element;
-    }
+    constructor(private readonly element: HTMLElement) {}
 
     public updateCursor(callerId: string, style?: string) {
-        delete this.states[callerId];
-
-        if (style != null) {
-            this.states[callerId] = { style };
-        }
-
-        this.applyStates();
-    }
-
-    private applyStates() {
-        let styleToApply = 'default';
-
-        // Last added entry wins.
-        Object.entries(this.states)
-            .reverse()
-            .slice(0, 1)
-            .forEach(([_, { style }]) => (styleToApply = style));
-
-        this.element.style.cursor = styleToApply;
+        this.stateTracker.set(callerId, style);
+        this.element.style.cursor = this.stateTracker.stateValue()!;
```

Comment:
oh checking again, I guess you could do `new StateTracker<string>('default')`

Reply:
I agree with you, tried to achieve something like that without any luck.
I can try again, maybe typing the constructor itself and what returns from it would work.

Reply:
So it seems that there no way to do this. The type must be declared *before* the constructor if called.

Reply:
doh, fair enough

Reply:
Had a quick play, and I think it is possible - but you need to make the default value required, and add `| undefined` to the generic type parameter at call-sites for other cases.